lib/vagrant-ovirt3/action/create_vm.rb

```diff
@@ -34,19 +34,30 @@ def call(env)
           env[:ovirt_cluster] = cluster
 
           # Get template
-          template = OVirtProvider::Util::Collection.find_matching(
-            env[:ovirt_compute].templates.all, config.template)
+          template = env[:ovirt_compute].templates.all.find_all { |t|
+            t.id == config.template or t.name == config.template
+          }
+          .sort_by { |t| t.raw.version.version_number.to_i }.reverse
+          .find { |t|
+            v = t.raw.version
+            cv = config.template_version
+            cv.nil? or (cv.to_i == v.version_number.to_i or cv == v.version_name)
+          }
           if template == nil
             raise Error::NoTemplateError,
               :template_name => config.template
           end
 
           # Output the settings we're going to use to the user
+          ver = template.raw.version
           env[:ui].info(I18n.t("vagrant_ovirt3.creating_vm"))
           env[:ui].info(" -- Name:          #{name}")
           env[:ui].info(" -- Cpus:          #{cpus}")
           env[:ui].info(" -- Memory:        #{memory_size/1024}M")
           env[:ui].info(" -- Template:      #{template.name}")
+          if ver.version_name or ver.version_number
+            env[:ui].info(" -- Version:       #{ver.version_name || ver.version_number}")
```

Comment:
I'm sorry, I missed these comments.  Both ideas seem good to me.  I will update with the more robust checks and the display change.